lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
+    return Object.keys(collectionSpecSubset).reduce((reductionResult, fieldName) =>
+      this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) && reductionResult
+    , true);
   }
 
-  if (documentSubset.hasOwnProperty(fieldName) && typeof documentSubset[fieldName] !== 'undefined' && documentSubset[fieldName] !== null) {
-    if (field.multivalued.value) {
-      if (!Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
-        return false;
-      }
-
-      if (field.multivalued.hasOwnProperty('minCount') && documentSubset[fieldName].length < field.multivalued.minCount) {
-        manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
-        return false;
-      }
-
-      if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
-        manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
-        return false;
-      }
+  /**
+   * @param {string} fieldName
+   * @param {*} documentSubset
+   * @param {StructuredFieldSpecification} collectionSpecSubset
+   * @param {boolean} strictness
+   * @param {string[]} errorMessages
+   * @param {boolean} verbose
+   * @returns {boolean}
+   */
+  isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
+    /** @type StructuredFieldSpecification */
+    const field = collectionSpecSubset[fieldName];
+    let result = true;
 
-      fieldValues = documentSubset[fieldName];
+    if (field.mandatory && !field.hasOwnProperty('defaultValue') && (!documentSubset.hasOwnProperty(fieldName) || typeof documentSubset[fieldName] === 'undefined' || documentSubset[fieldName] === null)) {
+      manageErrorMessage(field.path, errorMessages, 'The field is mandatory.', verbose);
+      return false;
     }
-    else {
-      if (Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field is not a multivalued field; Multiple values provided.', verbose);
-        return false;
-      }
 
-      fieldValues = [documentSubset[fieldName]];
-    }
+    if (documentSubset.hasOwnProperty(fieldName) && typeof documentSubset[fieldName] !== 'undefined' && documentSubset[fieldName] !== null) {
+      let
+        nestedStrictness = false,
+        fieldValues;
 
-    if (this.types[field.type].allowChildren) {
-      nestedStrictness = this.types[field.type].getStrictness(field.typeOptions, strictness);
-    }
+      if (field.multivalued.value) {
+        if (!Array.isArray(documentSubset[fieldName])) {
+          manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 121 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)